File: 1460-make-two-arrays-equal-by-reversing-subarrays/1460-make-two-arrays-equal-by-reversing-subarrays.cpp
class Solution {
public:
    bool canBeEqual(vector<int>& target, vector<int>& arr) {
        // Time O(n)
        // Space O(n)
        vector<int> mp(1001, 0);
        for(int i = 0; i<target.size(); i++){
            mp[target[i]] += 1;
            mp[arr[i]] -= 1;
        }
        for(int num : mp) if(num) return false;
        return true;
    }
};

File: 1460-make-two-arrays-equal-by-reversing-subarrays/1460-make-two-arrays-equal-by-reversing-subarrays.cs
// 4 2 3 5 1 
public class Solution {
    public bool CanBeEqual(int[] target, int[] arr) {
        int n = target.Length;
        int[] mp = new int[1001];
        
        for(int i = 0; i<n; i++){
            mp[target[i]] += 1;
            mp[arr[i]] -= 1;
        }
        
        foreach(int num in mp) if(num != 0) return false;
        return true;
    }
}